### PR TITLE
Add python program to measure cycle time of piBridge [REVPI-2787]

### PR DESCRIPTION
**Pull request by naruxde**
New script and documentation

### CODE REVIEW

File: pibridge_cycle_time/pibridge-cycle-time [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    """RevPiModIO measurement cycle at 10 ms."""
+    if ct.first:
+        ct.var.count = 0
+        ct.var.ts = {}
+
+    ct.var.count += 1
+    cycle_time = ct.io.RevPiIOCycle()
+    ct.var.ts[time()] = cycle_time
+
+    if not pargs.batch and (ct.first or not ct.var.count % 50):  # Refresh display on first run and every 500 ms
```
Comment (nbuchwitz) [inline pibridge_cycle_time/pibridge-cycle-time:66]: flake8: line too long (112 > 79 characters)

File: pibridge_cycle_time/pibridge-cycle-time [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        q_display.put_nowait(tuple(ct.var.ts.values()))  # Create a copy of dict values (mutable)
```
Comment (nbuchwitz) [inline pibridge_cycle_time/pibridge-cycle-time:67]: flake8: line too long (97 > 79 characters)

File: pibridge_cycle_time/pibridge-cycle-time [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+th_display.join()
+
+# Print JSON data to stdout
+dc_data = {
+    "time_local": datetime.now().isoformat(),
+    "duration_s": ct_result.var.count / 100,
+    "cycles": ct_result.var.count,
+    "min_ms": min(ct_result.var.ts.values()),
+    "max_ms": max(ct_result.var.ts.values()),
+    "mean_ms": round(sum(ct_result.var.ts.values()) / len(ct_result.var.ts.values()), 2),
```
Comment (nbuchwitz) [inline pibridge_cycle_time/pibridge-cycle-time:126]: flake8: line too long (89 > 79 characters)

File: pibridge_cycle_time/pibridge-cycle-time [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+import revpimodio2
+
+# Generate command arguments of the program
+parser = ArgumentParser(
+    prog="test_cycle_time",
+    formatter_class=RawDescriptionHelpFormatter,
+    description=textwrap.dedent("""\
+        Measure the cycle time of piControl to update all IO modules.
+
```
Comment (nbuchwitz) [inline pibridge_cycle_time/pibridge-cycle-time:26]: flake8: blank line contains whitespace

File: pibridge_cycle_time/pibridge-cycle-time [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+# Generate command arguments of the program
+parser = ArgumentParser(
+    prog="test_cycle_time",
+    formatter_class=RawDescriptionHelpFormatter,
+    description=textwrap.dedent("""\
+        Measure the cycle time of piControl to update all IO modules.
+        
+        This test is running in a 10 ms interval.
+
```
Comment (nbuchwitz) [inline pibridge_cycle_time/pibridge-cycle-time:28]: flake8: blank line contains whitespace

**Review by nbuchwitz — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# kpi-utils
+
+Collection of utilities for measuring key performance issues from the user's
```
Comment (iluminat23) [inline README.md:3]: key performance indicators

**Review by hjokunbus — COMMENTED** [review]
I generally suggest more expressive variable names. https://realpython.com/python-pep8/#how-to-choose-names

File: README.md [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# kpi-utils
```
[the remark this answers is not in the document]
Comment (naruxde) [reply README.md:1]: +1

File: pibridge_cycle_time/pibridge-cycle-time [hunk 8]
```diff
@@ -0,0 +1,144 @@
+#! /usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""Test the cycle time of Revolution Pi piControl."""
+__author__ = "Sven Sager"
+__copyright__ = "Copyright (C) KUNBUS GmbH, Heerweg 15C, 73770 Denkendorf, Germany"
```
Comment (iluminat23) [inline pibridge_cycle_time/pibridge-cycle-time:5]: Is "Copyright (C) KUNBUS GmbH" not enough? In our other packages we just use this

File: pibridge_cycle_time/pibridge-cycle-time [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+__license__ = "GPLv3"
```
Comment (iluminat23) [inline pibridge_cycle_time/pibridge-cycle-time:6]: IS GPLv3 mandatory? Normally we use GPLv2 or MIT as GPLv3 is problematic when used on embedded devices.

**Review by iluminat23 — APPROVED** [review]

**Pull request merged**